### PR TITLE
Area selector component

### PR DESCRIPTION
**Pull request by michaelkleinhenz**
This externalizes the selector typeahead dropdown and provides a seperate component for it. It also fixes several issues on the tests.

### CODE REVIEW

File: src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+.typeahead-dropdown {
+  padding-left: 0;
+    
+    .neutral-entry {
+      font-style: italic !important;
+      opacity: 0.5 !important;
+    }
+
+    .details-dropdown {
+      width: $width100;
```
Comment (mindreeper2420) [inline src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss]: just use `width: 100%;` this variable should be removed in the future.

File: src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      span {
+        z-index: 11;
+      }
+    }
+
+    .dropdown-search {
+      position: relative;
+      width: 40%;
+      height: 1px;
+      z-index: 10000;
```
Comment (mindreeper2420) [inline src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss]: why such a massive z-index? the dropdown in the navigation are only at 1050

File: src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        .item-ul {
+          background: $color-pf-white;
+          list-style: none;
+          margin-bottom: 0;
+          padding-left: em(10);
+
+          .item-li {
+            cursor: pointer;
+            padding: em(5);
+            width: $width100;
```
Comment (mindreeper2420) [inline src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss]: set to `width: 100%;`

File: src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<div class="col-sm-10 typeahead-dropdown pull-left">
+    <div *ngIf="!searchValue" (click)="open()" class="pointer details-dropdown" [ngClass]="getInitialValue().cssLabelClass?getInitialValue().cssLabelClass:''">{{ getInitialValue().value }}</div>
+    <div style="display: flex;">
```
Comment (mindreeper2420) [inline src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.html]: two things:
1. inline styles shouldn't be used
2. is it possible to do this without flex? flex will be removed post initial release

File: src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.html [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      <div *ngIf="searchValue" class="dropdown-search">
+        <div class="list-container">
+          <span class="fa fa-search dropdown-search-icon"> </span>
+          <input id="valueSearchInput" #valueSearch (keyup)="filterValue($event)" autocomplete="off" />
+          <ul class="item-ul dropdown-list" #valueList>
+            <li *ngIf="!values.length" class="textCenter"><i class="fa fa-spinner fa-pulse"></i></li>
```
Comment (mindreeper2420) [inline src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.html]: other areas of the UI use the PatternFly spinner (`<div class="spinner spinner-lg"></div>`). Size is changed by `spinner-lg / spinner-md / spinner-sm / spinner-xs`

File: src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+          width: $width100;
+          height: em(30);
+          border: none;
+          border-bottom: 1px solid $color-pf-black-500;
+          padding-left: em(20);
+        }
+
+        .dropdown-list {
+          max-height: em(150);
+          overflow-y: scroll;
```
Comment (mindreeper2420) [inline src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss]: default overflow for `overflow-x` is visible - will the dropdown list expand to accommodate larger text strings with a max-width?

File: src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+              font-weight: 600;
+            }
+          }
+
+          .selected {
+            background-color: $color-pf-black-300;
+          }
+
+          .action-li {
+            display: flex;
```
Comment (mindreeper2420) [inline src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss]: please remove flex, if able. custom `display: flex` properties will be removed post initial release throughout the product.

File: src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss (hunk 1 above)
Comment (michaelkleinhenz) [inline src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss]: Done.

File: src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss (hunk 2 above)
Comment (michaelkleinhenz) [inline src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss]: Mmmh..Just made sure it is on top :-)

File: src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss (hunk 6 above)
Comment (michaelkleinhenz) [inline src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss]: I don't know, I just moved that area from the detail view without changes. Let's keep it and put in a PR later on it.

File: src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss (hunk 7 above)
Comment (michaelkleinhenz) [inline src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.scss]: Yeah, I think this should be in @SMahil PR. There is a massive clash between those two PRs and I max have overwritten some of the work @SMahil did. We need to put that back in after this PR again.

File: src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.html (hunk 4 above)
Comment (michaelkleinhenz) [inline src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.html]: Same as above.

File: src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.html (hunk 5 above)
Comment (michaelkleinhenz) [inline src/app/work-item/work-item-detail/typeahead-dropdown/typeahead-dropdown.component.html]: Done.

**Comment by fabric8cd** [comment]
@michaelkleinhenz snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1618-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@michaelkleinhenz snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1618-fabric8-planner.int.rdu2c.fabric8.io

**Pull request merged**